ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -326,6 +326,9 @@ private void onRequestError(ExecutorToken ExecutorToken, int requestId, String e
     WritableArray args = Arguments.createArray();
     args.pushInt(requestId);
     args.pushString(error);
+    if (error.equals("timeout")) {
```

Comment:
Can you make this instead of using the string perhaps pass down the IOException? I'm not sure exactly what the exception type it'll be but perhaps `SocketTimeoutException`. Then just compare the `getClass()`. This avoids depending on an implementation detail somewhere setting "timeout" as the message.

It's probably easiest to split `onRequestError()` into two methods, one taking an error string and one taking an exception.